### DISCUSSION TITLE
Is there a network request for a Sveltekit endpoint during SSR.

### DISCUSSION THREAD

**Discussion by itssumitrai** [sveltejs/kit#5518] [Q&A] [answered]
Hi,

I have my `load` function something like for my `index.svelte`

```js
<script context="module">
    export const prerender = false;
    export async function load({ fetch }) {
        const data = await fetch('/xhr/foo');

        return {
            props: { data }
        };
    }
</script>
```

`load` function would run 2 times once on the Server & again on the client. My question is since its a relative url `/xhr/foo`, its another route defined in Sveltekit under `/src/routes/xhr/foo.js`.
Would there be an actual network request to itself for this endpoint on Server or Sveltekit is smart enough to not make a network request & execute the endpoint on server ?

Thanks.

**Comment by geoffrich** [comment, marked as answer]
Per the [docs](https://kit.svelte.dev/docs/loading#input-fetch), when using the `fetch` provided by SvelteKit:

> requests for endpoints go direct to the handler function during server-side rendering, without the overhead of an HTTP call

So the endpoint will be executed on the server without making another HTTP call.

**Reply by itssumitrai** [reply]
Awesome, thanks for reply
